zend_operator.go: remove unnecessary Floats interface

moonjihae opened the pull request:
#24
- The `Floats` interface was exported as public but was unnecessary in the codebase. 
This commit removes the interface to improve type safety and maintain clean code practices.
 
- Refactor `floatToString` function to accept only float64
    - Changed the parameter type of the `floatToString` function from
  `float32 | float64` to `float64`.
    - Updated the `ConvertToString`function to convert `float32` variable to
  `float64` before calling the`floatToString` function.
    - Added a note comment in the `ConvertToString` function documentation
  about precision loss when converting from `float32` to `float64`.

moonjihae committed:
zend_operator.go: remove unnecessary Floats interface

The Floats interface was exported as public but was unnecessary
in the codebase. This commit removes the interface to improve
type safety and maintain clean code practices.

simnalamburt commented on zend_operator.go:
```diff
@@ -32,7 +28,7 @@ type Floats interface {
 // Reference :
 //   - https://github.com/php/php-src/blob/php-5.6.40/Zend/zend_operators.c#L627-L633
-func floatToString[T Floats](value T) string {
+func floatToString[T float32 | float64](value T) string {
 	f64 := float64(value)
```
Why don't we just change the type parameter into float64 (and let caller do the float32 -> float64 casting)?

```suggestion
func floatToString(f64 float64) string {
```

moonjihae replied:
Refactored to accept only float64 and Updated the `ConvertToString`function to convert float32 variable to
float64 before calling the `floatToString` function.

moonjihae committed:
zend_operator.go: Refactor floatToString function to accept only float64

- Changed the parameter type of the `floatToString` function from
  `float32 | float64` to `float64`.
- Updated the `ConvertToString`function to convert `float32` variable to
  `float64` before calling the`floatToString` function.
- Added a note comment in the `ConvertToString` function documentation
  about precision loss when converting from `float32` to `float64`.
